beacon-chain/state/stategen/replay.go

```diff
@@ -16,6 +16,52 @@ import (
 	"go.opencensus.io/trace"
 )
 
+// ComputeStateUpToSlot returns a processed state up to input target slot.
+// If the last processed block is at slot 32, given input target slot at 40, this
+// returns processed state up to slot 40 via empty slots.
+func (s *State) ComputeStateUpToSlot(ctx context.Context, targetSlot uint64) (*state.BeaconState, error) {
```

Comment:
Ok , what if you have two blocks at slot 30 ? how would you know which state to return ?

Reply:
It returns the canonical one (ie ancestor of the canonical chain from the node's point of view)
Will update the comment